app/views/layouts/application.html.erb

```diff
@@ -58,6 +53,8 @@
     <%= yield :head %>
 
     <%= build_plugin_html 'server:before-head-close' %>
+
+    <%= tag.meta id: 'data-discourse-setup', data: client_side_setup_data %>
```

Comment:
Rebased and kept this in `<head>` since it's a lot smaller and a lot more visually pleasing than `preloaded-data`:

```html
<meta id="data-discourse-setup" data-base-url="localhost" data-base-uri="" 
data-environment="development" data-letter-avatar-version="5_06ddb670fcf086fa527f7887a59f15f8" 
data-markdown-it-url="http://localhost:3000/assets/markdown-it-bundle.js" 
data-service-worker-url="service-worker.js" data-default-locale="en" 
data-asset-version="d41d8cd98f00b204e9800998ecf8427e" 
data-disable-custom-css="false" data-highlight-js-path="/highlight-js/localhost/100444ceef8ca161334f64d3ce49fe7a1556f943.js">
```

(and just a tiny bit longer in production with S3)

Let's merge on Monday tho if all else is 👍